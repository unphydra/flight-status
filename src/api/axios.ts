import axios from "axios";

export default axios.create({
  baseURL: "https://flight-status-mock.core.travelopia.cloud",
});
