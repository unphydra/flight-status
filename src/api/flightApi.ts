import { AxiosInstance } from "axios";

export enum FlightStatus {
  ON_TIME = "On Time",
  BOARDING = "Boarding",
  DELAYED = "Delayed",
  DEPARTED = "Departed",
}

export type FlightDetail = {
  id: number;
  flightNumber: string;
  airline: string;
  origin: string;
  destination: string;
  departureTime: string;
  status: FlightStatus;
};

export default class FlightApi {
  constructor(private apiService: AxiosInstance) {}

  async fetchFlights(): Promise<FlightDetail[]> {
    const { data } = await this.apiService.get("/flights");
    return data;
  }
}
