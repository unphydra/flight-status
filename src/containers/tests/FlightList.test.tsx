import { render, screen, waitFor } from "@testing-library/react";
import { FlightDetail, FlightStatus } from "../../api/flightApi";
import FlightListContainer from "../FlightList";

const mocks = vi.hoisted(() => ({
  get: vi.fn(),
}));

vi.mock("axios", async (importActual) => {
  const actual = await importActual<typeof import("axios")>();

  const mockAxios = {
    default: {
      ...actual.default,
      create: vi.fn(() => ({
        ...actual.default.create(),
        get: mocks.get,
      })),
    },
  };

  return mockAxios;
});

vi.mock("../../components/flightTable", () => ({
  default: () => <div data-testid="flight-table"></div>,
}));

describe("FlightListContainer", () => {
  const flightDetails: FlightDetail = {
    id: 1,
    flightNumber: "1",
    airline: "airline",
    origin: "origin",
    destination: "destination",
    departureTime: "departureTime",
    status: FlightStatus.ON_TIME,
  };

  beforeEach(() => {
    vi.resetAllMocks();
  });

  it("Should renders flight table", async () => {
    mocks.get.mockResolvedValue({ data: [flightDetails] });
    render(<FlightListContainer />);
    await waitFor(() => {
      const flightTable = screen.getByTestId("flight-table");
      expect(flightTable).toBeInTheDocument();
    });
  });
});
