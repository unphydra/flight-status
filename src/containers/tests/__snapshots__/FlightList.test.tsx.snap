// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FlightListContainer > Should renders and match snapshot 1`] = `
<div>
  <div
    data-testid="flight-table"
  />
</div>
`;
